### PR TITLE
Update for Swift 4 syntax and use pure Swift/non-ObjC code

### PR DESCRIPTION
**Pull request by the-braveknight**
- Updated the code for Swift 4 syntax
- Use pure Swift code

### CODE REVIEW

File: Blog_NSNotificationCenter.playground/Contents.swift [hunk 1]
```diff
@@ -10,55 +10,38 @@ public extension Notifier where Notification.RawValue == String {
     
     // MARK: - Static Computed Variables
     
-    private static func nameFor(notification: Notification) -> String {
-        return "\(self).\(notification.rawValue)"
+    private static func name(forNotification notification: Notification) -> NSNotification.Name {
```
Comment (andyyhope) [inline Blog_NSNotificationCenter.playground/Contents.swift]: This doesnt align with Swift API guidelines, it should be 
`name(for notification: Notification)`

**Comment by the-braveknight** [comment]
Renamed name(for:) function.

File: Blog_NSNotificationCenter.playground/Contents.swift (hunk 1 above)
Comment (the-braveknight) [reply Blog_NSNotificationCenter.playground/Contents.swift]: Yes, you're right.

**Comment by the-braveknight** [comment]
Added few more changes to match what NotificationCenter class functions expect

**Pull request closed without merging**